apiserver/internal/handlers/objects/objects.go

```diff
@@ -8,391 +8,102 @@ import (
-	"github.com/juju/juju/domain/application"
-	"github.com/juju/juju/domain/application/architecture"
-	applicationcharm "github.com/juju/juju/domain/application/charm"
-	applicationerrors "github.com/juju/juju/domain/application/errors"
-	"github.com/juju/juju/internal/charm"
+	objectstoreerrors "github.com/juju/juju/domain/objectstore/errors"
 	"github.com/juju/juju/internal/errors"
-	internallogger "github.com/juju/juju/internal/logger"
-	"github.com/juju/juju/rpc/params"
-	"github.com/juju/juju/state"
 )
 
-var (
-	logger = internallogger.GetLogger("juju.apiserver.objects")
-)
-
-const (
-	// maxUploadSize is the maximum size of a charm that can be uploaded.
-	// TODO (stickupkid): This should be configurable either by a model config
-	// or a controller config.
-	// This number was derived from the max size of a charm in the charmhub.
-	// There are a few larger charms, but they're corrupted (charms within
-	// charms - inception!) and should be considered outliers.
-	// It's better to have an upper limit rather than no limit at all.
-	maxUploadSize = 500 * 1024 * 1024 // 500MB
-
-	// uploadTimeout is the maximum time allowed for a single charm upload.
-	// TODO (stickupkid): This should be configurable either by a model config
-	// or a controller config.
-	uploadTimeout = time.Minute * 5
-)
-
-// StateGetter is an interface for getting the model state.
-type StateGetter interface {
-	GetState(*http.Request) (ModelState, error)
+// ObjectStoreService is an interface that provides a method to get an object
+// from an object store.
+type ObjectStoreService interface {
+	// GetBySHA256 returns a reader for the object with the given SHA256 hash.
+	GetBySHA256(ctx context.Context, sha256 string) (io.ReadCloser, int64, error)
 }
 
-// ApplicationService is an interface for the application domain service.
-type ApplicationService interface {
-	// GetCharmArchiveBySHA256Prefix returns a ReadCloser stream for the charm
-	// archive who's SHA256 hash starts with the provided prefix.
-	//
-	// If the charm does not exist, a [applicationerrors.CharmNotFound] error is
-	// returned.
-	GetCharmArchiveBySHA256Prefix(ctx context.Context, sha256Prefix string) (io.ReadCloser, error)
-
-	// ResolveUploadCharm resolves the upload of a charm archive.
-	ResolveUploadCharm(context.Context, applicationcharm.ResolveUploadCharm) (applicationcharm.CharmLocator, error)
+// ObjectStoreServiceGetter is an interface that provides a method to get an
+// object store service.
+type ObjectStoreServiceGetter interface {
+	ObjectStore(*http.Request) (ObjectStoreService, error)
 }
 
-// ApplicationServiceGetter is an interface for getting an ApplicationService.
-type ApplicationServiceGetter interface {
-	// Application returns the model's application service.
-	Application(*http.Request) (ApplicationService, error)
+// ObjectsHTTPHandler implements the http.Handler interface for the objects API.
+type ObjectsHTTPHandler struct {
+	stateGetter       StateGetter
```

Comment:
`stateGetter` is unused on this handler!

Reply:
Noice. Thanks for the spot.